models/baseModels/Invoice/Invoice.ts

```diff
@@ -162,16 +172,44 @@ export abstract class Invoice extends Transactional {
     await validateBatch(this);
   }
 
+  async getNewBaseGrandTotal() {
+    const totalLotaltyAmount = await getNewGrandTotal(
+      this.loyaltyProgram as string,
+      this.loyaltyPoints as number
+    );
+
+    return totalLotaltyAmount.sub(this.baseGrandTotal as Money).abs();
+  }
+
   async afterSubmit() {
     await super.afterSubmit();
 
+    if (this.isReturn) {
+      console.log(this, 'THIS');
+
+      await this._removeLoyaltyPointEntry();
+    }
+
+    if (this.schemaName === ModelNameEnum.SalesQuote) {
+      return;
+    }
+    let newBaseGrandTotal: Money | undefined;
+
+    if (this.redeemLoyaltyPoints) {
+      newBaseGrandTotal = await this.getNewBaseGrandTotal();
```

Comment:
the variable name `newBaseGrandTotal` needs to be changed to something that matches its purpose